### DISCUSSION TITLE
Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: anthropic-claude-3-5-sonnet

### DISCUSSION THREAD

**Discussion by mymaher** [microsoft/semantic-kernel#7286] [Q&A] [answered]
Hello everyone,
Need help to resolve below exception. It may be a pre-release bug in (Microsoft.SemanticKernel.Connectors.Google)

Same code is working fine with AzureOpenAI

Exception:
{
  "error": {
    "code": 429,
    "message": "Quota exceeded for aiplatform.googleapis.com/generate_content_requests_per_minute_per_project_per_base_model with base model: anthropic-claude-3-5-sonnet. Please submit a quota increase request. https://cloud.google.com/vertex-ai/docs/generative-ai/quotas-genai.",
    "status": "RESOURCE_EXHAUSTED"
  }
}

C# Code:
var builder = Host.CreateEmptyApplicationBuilder(new HostApplicationBuilderSettings { Args = args });

#pragma warning disable SKEXP0070
builder.Services.AddVertexAIGeminiChatCompletion(modelId, async () => { return await GetToken(); },location, projectId);

var app = builder.Build();

var chatHistory = new ChatHistory();
var chat = app.Services.GetRequiredService<IChatCompletionService>();

chatHistory.AddUserMessage("Hi");

var lastMessage = await chat.GetChatMessageContentAsync(chatHistory); //exception raised on this line

![Exception1](https://github.com/user-attachments/assets/09694dcf-f9f5-43cb-81ef-238da03666f3)

![Exception2](https://github.com/user-attachments/assets/6f42b6c8-e348-4c70-8e24-00911886661d)

**Comment by sophialagerkranspandey** [comment]
The error code you've shared is related to the limit request of the connector you're using. We would recommend you look at the limits of your resource.

**Reply by mymaher** [reply]
Dear Sophia,

Thank you for your reply.

Using Postman, I am able to call the Google API without any issues. However, when I use my console application with AddVertexAIGeminiChatCompletion, I cannot make a single call. I reviewed the Google quota documentation, and we are within the limit.

Any suggestions would be much appreciated.

**Reply by sophialagerkranspandey** [reply, marked as answer]
Are you seeing a different request payload from Semantic Kernel then what you seeing using Postman? You can do this using a custom HTTP handler and putting a breakpoint in it.

**Reply by mymaher** [reply]
I cloned the [Semantic Kernel](https://github.com/microsoft/semantic-kernel) project and included it as a reference to debug thoroughly. I was using the model 'anthropic-claude-3-5-sonnet,' which has a different payload compared to the implemented Gemini model. Additionally, I noticed an infinite for loop (I believe loop will break later based on some condition). However, it looks like that due to different payload exception, it keeps calling the API and causing a quota exceeded exception and hiding actual exception of bad request.

Due to time constraints, I am now doing custom implementation.

**Reply by Vifill** [reply]
I also have the same issue.  @mymaher Do you have the custom implementation available somewhere?

@sophialagerkranspandey Do you know if this is known and will be fied?
